From 54a3d5e1cd0ebe634192da71c76ba129c420ed43 Mon Sep 17 00:00:00 2001
From: Mirela Rabulea <mirela.rabulea@nxp.com>
Date: Wed, 28 Aug 2019 14:08:47 +0300
Subject: [PATCH 01/37] MLK-22592: mxc-jpeg: Fix v4l2-compliance, "Control
 ioctls", UNSUBSCRIBE_EVENT

commit acf79f46d77eb628b41a5db05adf129811319004 from
https://source.codeaurora.org/external/imx/linux-imx.git

Fail: from "Control ioctls", test VIDIOC_(UN)SUBSCRIBE_EVENT/DQEVENT
v4l2-test-controls.cpp(886): doioctl(node, VIDIOC_UNSUBSCRIBE_EVENT, &sub)

Fix: add default v4l2 implementation for vidioc_unsubscribe_event ioctl

Signed-off-by: Mirela Rabulea <mirela.rabulea@nxp.com>
Reviewed-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Reviewed-by: Robert Chiras <robert.chiras@nxp.com>
(cherry picked from commit fa547b9bd2eef0a7966d406653a26ffb31da097a)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-jpeg.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/platform/imx8/mxc-jpeg.c b/drivers/media/platform/imx8/mxc-jpeg.c
index 85a5a1e8e395..69397a19b298 100644
--- a/drivers/media/platform/imx8/mxc-jpeg.c
+++ b/drivers/media/platform/imx8/mxc-jpeg.c
@@ -1882,6 +1882,7 @@ static const struct v4l2_ioctl_ops mxc_jpeg_ioctl_ops = {
 	.vidioc_g_fmt_vid_out_mplane		= mxc_jpeg_g_fmt_vid_out,
 
 	.vidioc_subscribe_event		= mxc_jpeg_subscribe_event,
+	.vidioc_unsubscribe_event	= v4l2_event_unsubscribe,
 	.vidioc_decoder_cmd		= mxc_jpeg_decoder_cmd,
 	.vidioc_encoder_cmd		= mxc_jpeg_encoder_cmd,
 
-- 
2.17.1

